PWGHF/TableProducer/candidateSelectorXicToXiPiPi.cxx

```diff
@@ -0,0 +1,247 @@
+      selectorPion.setRangeNSigmaTpcCondTof(-nSigmaTpcCombinedMax, nSigmaTpcCombinedMax);
+      selectorPion.setRangePtTof(ptPidTofMin, ptPidTofMax);
+      selectorPion.setRangeNSigmaTof(-nSigmaTofMax, nSigmaTofMax);
+      selectorPion.setRangeNSigmaTofCondTpc(-nSigmaTofCombinedMax, nSigmaTofCombinedMax);
+      // proton
+      selectorProton.setRangePtTpc(ptPidTpcMin, ptPidTpcMax);
+      selectorProton.setRangeNSigmaTpc(-nSigmaTpcMax, nSigmaTpcMax);
+      selectorProton.setRangeNSigmaTpcCondTof(-nSigmaTpcCombinedMax, nSigmaTpcCombinedMax);
+      selectorProton.setRangePtTof(ptPidTofMin, ptPidTofMax);
+      selectorProton.setRangeNSigmaTof(-nSigmaTofMax, nSigmaTofMax);
+      selectorProton.setRangeNSigmaTofCondTpc(-nSigmaTofCombinedMax, nSigmaTofCombinedMax);
+    }
+
+    if (activateQA) {
+      constexpr int kNBinsSelections = 1 + SelectionStep::NSelectionSteps;
+      std::string labels[kNBinsSelections];
+      labels[0] = "No selection";
+      labels[1 + SelectionStep::RecoSkims] = "Skims selection";
+      labels[1 + SelectionStep::RecoTopol] = "Skims & Topological selections";
+      labels[1 + SelectionStep::RecoPID] = "Skims & Topological & PID selections";
+      labels[1 + SelectionStep::RecoMl] = "Skims & Topological & PID & ML selections";
+      static const AxisSpec axisSelections = {kNBinsSelections, 0.5, kNBinsSelections + 0.5, ""};
+      registry.add("hSelections", "Selections;;#it{p}_{T} (GeV/#it{c})", {HistType::kTH2F, {axisSelections, {(std::vector<double>)binsPt, "#it{p}_{T} (GeV/#it{c})"}}});
+      for (int iBin = 0; iBin < kNBinsSelections; ++iBin) {
+        registry.get<TH2>(HIST("hSelections"))->GetXaxis()->SetBinLabel(iBin + 1, labels[iBin].data());
+      }
+    }
+  }
+
+  /// Conjugate-independent topological cuts
+  /// \param candidate is candidate
+  /// \return true if candidate passes all cuts
+  template <typename T1>
+  bool selectionTopol(const T1& hfCandXic)
+  {
+    auto candpT = hfCandXic.pt();
+    int pTBin = findBin(binsPt, candpT);
+    if (pTBin == -1) {
+      return false;
+    }
+
+    // check that the candidate pT is within the analysis range
+    if (candpT < ptCandMin || candpT >= ptCandMax) {
+      return false;
+    }
+
+    // check candidate mass is within a defined mass window
+    if (std::abs(hfCandXic.invMassXic() - o2::constants::physics::MassXiCPlus) > cuts->get(pTBin, "m")) {
+      return false;
+    }
+
+    // cosine of pointing angle
+    if (hfCandXic.cpa() <= cuts->get(pTBin, "cos pointing angle")) {
+      return false;
+    }
+
+    // cosine of pointing angle XY
+    if (hfCandXic.cpaXY() <= cuts->get(pTBin, "cos pointing angle XY")) {
+      return false;
+    }
+
+    // candidate maximum decay length
+    if (hfCandXic.decayLength() > cuts->get(pTBin, "max decay length")) {
+      return false;
+    }
+
+    // candidate maximum decay length XY
+    if (hfCandXic.decayLengthXY() > cuts->get(pTBin, "max decay length XY")) {
+      return false;
+    }
```

Comment:
2 questions here:
 - is this upper cut intended to suppress the beauty component?
 - you have a 50 micron cut on this variable. Are you sure that this is not too tight for a secondary vertex built with from a cascade?

Reply:
To be honest, I just started from the Bs selector that I wrote as well and didn't think about it too much. If I didn't overlook something, the XicPlus has feeddown contributions only from the OmegaBMinus. Do you think this cut is useful in any way?
When I tested my code locally, I also set different values for this cut since it basically kills all the candidates as you mentioned. I modified the default values in the SelectorCuts.h accordingly.